src/assets/scripts/aircraft/AircraftFlightManagementSystem.js

```diff
@@ -180,15 +206,17 @@ export default class AircraftFlightManagementSystem {
      *  Insert a waypoint after the *current* waypoint
      */
     appendWaypoint(data) {
-        this.currentLeg().waypoints.splice(this.current[WAYPOINT_WITHIN_LEG] + 1, 0, new Waypoint(data, this));
+        const airport = window.airportController.airport_get();
```

Comment:
At some point, we might want to change `airport_get()` to a getter. Just a thought.